src/gui/AutomatableModelView.cpp

```diff
@@ -258,7 +287,251 @@ void AutomatableModelViewSlots::removeConnection()
+	AutomationNodeAtTimePos nodeClip = getNearestAutomationNode(track);
+	if (nodeClip.m_clip)
+	{
+		// modifying its value
+		nodeClip.m_clip->recordValue(nodeClip.m_position, m_amv->modelUntyped()->getRawValueOrControllerValue());
+	}
+}
+
+void AutomatableModelViewSlots::removeSongNearestAutomationNode()
+{
+	// getting the track without adding a new one if no track was found
+	AutomationTrack* track = getCurrentAutomationTrackForModel(false);
+
+	// this needs to be checked because getCurrentAutomationTrack might give
+	// a nullptr if it can not find and add a track
+	if (!track) { return; }
+
+	AutomationNodeAtTimePos nodeClip = getNearestAutomationNode(track);
+	if (nodeClip.m_clip)
+	{
+		nodeClip.m_clip->removeNode(nodeClip.m_position);
+		// if there is no node left, the automationClip will be deleted
+		if (nodeClip.m_clip->hasAutomation() == false)
+		{
+			// this is safe because clips remove themself from tracks
+			delete nodeClip.m_clip;
+		}
+	}
+}
+
+AutomationTrack* AutomatableModelViewSlots::getCurrentAutomationTrack(std::vector<AutomationClip*>* clips, bool canAddNewTrack)
+{
+	AutomationTrack* output = nullptr;
+	if (clips->size() > 0)
+	{
+		// selecting the track with the most amount of clips
+		// connected to this model
+
+		// track*, how many clips are on that track (that are connected to this model)
+		std::list<std::pair<AutomationTrack*, size_t>> trackList;
+		for (size_t i = 0; i < clips->size(); i++)
+		{
+			bool found = false;
+			// search this track inside the existing tracks
+			for (std::pair<AutomationTrack*, size_t>& j : trackList)
+			{
+				// if the track already is in trackList
+				if (j.first == (*clips)[i]->getTrack())
+				{
+					found = true;
+					j.second++;
+					break;
+				}
+			}
+			if (!found)
+			{
+				trackList.push_back(std::make_pair(dynamic_cast<AutomationTrack*>((*clips)[i]->getTrack()), 1));
+			}
+		}
+
+		size_t matchedModelCount = 0;
+		AutomationTrack* matchedTrack = nullptr;
+		// find a track where all clips are all connected to this model
+		for (std::pair<AutomationTrack*, size_t>& j : trackList)
+		{
+			// if all clips on that track are connected to this model
+			bool isOnlyThatModel = static_cast<size_t>(j.first->numOfClips()) == j.second;
+			if (matchedModelCount < j.second && isOnlyThatModel)
+			{
+				matchedTrack = j.first;
+				matchedModelCount = j.second;
+			}
+		}
 
+		output = matchedTrack;
+	}
+	if (canAddNewTrack && output == nullptr)
+	{
+		// adding new track
+		output = new AutomationTrack(getGUI()->songEditor()->m_editor->model(), false);
+		output->setName(m_amv->modelUntyped()->displayName() + " " + tr("automation"));
+	}
+	return output;
+}
+
+AutomationClip* AutomatableModelViewSlots::getCurrentAutomationClip(AutomationTrack* track, bool canAddNewClip, bool searchAfter)
+{
+	AutomationClip* output = nullptr;
+	const std::vector<Clip*>& trackClips = track->getClips();
+	TimePos timePos = static_cast<TimePos>(Engine::getSong()->getPlayPos());
```

Comment:
This rather technical code with the cast repeats several times in this PR. Might be worth to isolate it in a private helper method, e.g. `getCurrentPlayingPosition()`. ("Don't repeat yourself" principle)